src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -86,6 +88,43 @@ public AWSResourceQueryer(IAWSClientFactory awsClientFactory)
             _awsClientFactory = awsClientFactory;
         }
 
+        public async Task<List<Subnet>> DescribeSubnets(List<string>? subnetIds = null)
+        {
+            var ec2Client = _awsClientFactory.GetAWSClient<IAmazonEC2>();
+
+            var request = new DescribeSubnetsRequest();
+            if (subnetIds != null)
+                request.SubnetIds = subnetIds;
+
+            return await ec2Client.Paginators
+                .DescribeSubnets(request)
+                .Subnets
+                .ToListAsync();
+        }
+
+        public async Task<List<SecurityGroup>> DescribeSecurityGroups(List<string>? subnetIds = null)
+        {
+            var ec2Client = _awsClientFactory.GetAWSClient<IAmazonEC2>();
+            var request = new DescribeSecurityGroupsRequest();
+            if (subnetIds != null)
+            {
+                var subnets = await DescribeSubnets(subnetIds);
+                request.Filters = new List<Filter>()
+                {
+                    new Filter()
```

Comment:
You should only add the filter if there is a subnet. Either that or don't allow the call if there are no subnets.

Reply:
Done